### ISSUE TITLE
Aplication not found WindowsDesktop. @area-Host

### ISSUE THREAD

**Issue by michalkress** [dotnet/runtime#67580]
### Description

Hi,
A created WPF application in .core (6.0.1) and try run from release bin folder (SDK version 6.0.3). I have a problem with Deskop Runtime:
![image](https://user-images.githubusercontent.com/19835872/161732343-48dee287-d118-4d35-96e8-1e695b3514c9.png)

I have Installed frameworks on my machine:
![image](https://user-images.githubusercontent.com/19835872/161731385-ea35f8f3-3a08-4544-8767-2d0dda5a732d.png)

I don't have Desktop Runtime installed i 6.0.2 version. I haven't idea for solution this problem. I trying using RollForward, RuntimeFrameworkVersion.

When I uninstall Runtime 6.0.2, application works, but some my client meybe have this problem in future, when try run my application.

Runtime json configuration:
`{
  "runtimeOptions": {
    "tfm": "net6.0",
    "frameworks": [
      {
        "name": "Microsoft.NETCore.App",
        "version": "6.0.1",	
      },
      {
        "name": "Microsoft.WindowsDesktop.App",
        "version": "6.0.1",
      }
    ],
    "configProperties": {
      "System.Globalization.Invariant": false,
      "System.Reflection.Metadata.MetadataUpdater.IsSupported": true
    }
  }
}
`
Project configuration:
`
<TargetFramework>net6.0-windows</TargetFramework>
    <OutputType>Exe</OutputType>
    <UseWPF>true</UseWPF>
    <UseWindowsForms>true</UseWindowsForms>
    <RuntimeIdentifiers>win-x86</RuntimeIdentifiers>    
    <PuglishSingleFile>true</PuglishSingleFile>
    <EnableDynamicLoading>true</EnableDynamicLoading>
    <GenerateRuntimeConfigurationFiles>true</GenerateRuntimeConfigurationFiles>
    <RuntimeFrameworkVersion>6.0.1</RuntimeFrameworkVersion>
`

### Reproduction Steps

- Application in WPF (.net core 6.01)
- Installed .NET Desktop Runtime 6.0.1 and .NET Runtime 6.0.2
- Run an Application in WPF.

### Expected behavior

when I run an application that requires two components, I run it in an environment of two components, not the newest one.

### Actual behavior

when I run an application that requires two components, I run it in last version runtime, but not check desktop runtime is exist.

**Comment by vitek-karas** [comment]
What was the version of the SDK you used to build the app?

I'm able to repro this if I use 6.0.300 SDK to build the app, and then run it on a setup where only 6.0.1 Microsoft.Windows.Desktop is installed - then it fails since it's looking for `PresenationFramework` version 6.0.2. Note that the `app.dll` produced by the SDK has an assembly reference to `PresentationFramework` version 6.0.2. So the app effectively requires Microsoft.Windows.Desktop framework version 6.0.2 or higher. But the `.runtimeconfig.json` lists the versions as `6.0.0`.

@marcpopMSFT This feels more like an SDK question - basically what are the expectations on an app produced by SDK 6.0.300 - should it run on 6.0.1 runtimes, or does it require 6.0.2 at least.

When I build the app with 6.0.100 SDK, then it works just fine on 6.0.1 runtimes.

**Comment by michalkress** [comment]
I using 6.0.200 version, but I now tested 6.0.101 and works fine. 

Now I created `global.json` for select SDK but this resolve is not a great solution for the future.

I think: 
- When `PresentationFramework` in version 6.0.2 doesn't exist, the system should show a missing message about framework.
- Users of programs on the world are not very smart and likes experiments with an elements of system (removing, installing, changing, runs unknown applications, ....) - Today I had a problem with framework on testing machine ;).
- How can I check and notified problem with `PresentationFramework` at my customer?
- How to respond when application encounter this problem?
